tests/ui/attributes/linkage.rs

```diff
@@ -0,0 +1,38 @@
+#![feature(linkage)]
+#![feature(stmt_expr_attributes)]
+#![deny(unused_attributes)]
+#![allow(dead_code)]
+
```

Comment:
based on the error message I thinks these would be good test cases

```rust
trait G {
    #[linkage = "weak"]
    fn foo();
}

extern "C" {
    #[linkage = "weak"]
    fn bar();
}
```

these look like functions, but are not function definitions